src/new-apps/components/AppDetailsPage/HeaderOptions.tsx

```diff
@@ -0,0 +1,77 @@
+import ExternalLink from "@dashboard/components/ExternalLink";
+import Skeleton from "@dashboard/components/Skeleton";
+import { AppQuery } from "@dashboard/graphql";
+import { buttonMessages } from "@dashboard/intl";
+import { ButtonBase } from "@material-ui/core";
+import { Box } from "@saleor/macaw-ui/next";
+import React from "react";
+import SVG from "react-inlinesvg";
+import { FormattedMessage } from "react-intl";
+
+import activateIcon from "../../../../assets/images/activate-icon.svg";
+import deleteIcon from "../../../../assets/images/delete.svg";
+import supportIcon from "../../../../assets/images/support-icon.svg";
+import messages from "./messages";
+import { useStyles } from "./styles";
+
+interface HeaderOptionsProps {
+  data: AppQuery["app"];
+  onAppActivateOpen: () => void;
+  onAppDeactivateOpen: () => void;
+  onAppDeleteOpen: () => void;
+}
+
+const HeaderOptions: React.FC<HeaderOptionsProps> = ({
+  data,
+  onAppActivateOpen,
+  onAppDeactivateOpen,
+  onAppDeleteOpen,
+}) => {
+  const classes = useStyles();
+
+  if (!data) {
```

Comment:
**issue**: Props don't suggest data can be undefined. Perhaps it would be a good idea to run strict null check on these files

Reply:
Prop type is equal to `AppQuery["app"]`, which is object or null.